src/index.js

```diff
@@ -1 +1,54 @@
-console.log('Happy developing ✨')
+import './style.css';
+import { getImageElementWhenLoaded } from './components/getImageElementWhenLoaded';
+import { wait } from './components/wait';
+
+const images = [
+  'https://picsum.photos/5000?random=1',
+  'https://picsum.photos/5000?random=2',
+  'https://picsum.photos/5000?random=3',
+  'https://picsum.photos/5000?random=4',
+];
+
+const buttonsWrapper = document.querySelector('.buttons-wrapper');
+const informationParagraph = document.querySelector('.loading-info');
+
+images.forEach(function (image) {
+  const button = document.createElement('button');
+  button.classList.add('button');
+  button.innerText = images.indexOf(image) + 1;
+  buttonsWrapper.appendChild(button);
+
+  button.addEventListener('click', function () {
+    const buttonIndex = Array.from(button.parentNode.children).indexOf(button);
+    const selectedImage = images[buttonIndex];
+    const imageWrapper = document.querySelector('.image-wrapper');
+    informationParagraph.innerText = 'loading...';
+
+    getImageElementWhenLoaded(selectedImage, imageWrapper)
+      .then(function (image) {
+        const currentImage = imageWrapper.querySelector('img');
+
+        if (currentImage) {
+          currentImage.classList.remove('loaded');
+          wait(700).then(function () {
+            currentImage.remove();
+          });
+          wait(700).then(function () {
+            imageWrapper.append(image);
+          });
```

Comment:
You are doing `wait(700)` twice which does not make much sense. You could simplify it to
```javascript
wait(700).then(function () {
  currentImage.remove();
  imageWrapper.append(image);
});
```

Reply:
We can simplify it even more with chaining
by first waiting for 700 milliseconds and then waiting for 10 more milliseconds
```javascript
getImageElementWhenLoaded(selectedImage, imageWrapper)
  .then(function (image) {
    const currentImage = imageWrapper.querySelector('img');

    if (currentImage) {
      currentImage.classList.remove('loaded');
      wait(700)
        .then(function () {
          currentImage.remove();
          imageWrapper.append(image);
          return wait(10);
        })
        .then(function() {
          image.classList.add('loaded');
        })
    } else {
      imageWrapper.append(image);
      wait(10).then(function () {
        image.classList.add('loaded');
      });
    }
    informationParagraph.innerText = '';
  })
  .catch(function () {
    informationParagraph.innerText = 'loading error';
  });
```

Reply:
We could simplify it further by changing the order of if statements and using `return`

```javascript
getImageElementWhenLoaded(selectedImage, imageWrapper)
  .then(function (image) {
    const currentImage = imageWrapper.querySelector('img');
    informationParagraph.innerText = '';

    if (!currentImage) {
      imageWrapper.append(image);
      return wait(10)
        .then(function () {
          image.classList.add('loaded');
        });
    }

    currentImage.classList.remove('loaded');
    return wait(700)
      .then(function () {
        currentImage.remove();
        imageWrapper.append(image);
        return wait(10);
      })
      .then(function() {
        image.classList.add('loaded');
      })
  })
  .catch(function () {
    informationParagraph.innerText = 'loading error';
  });
```